include/accessor.h

```diff
@@ -261,44 +286,81 @@ class accessor<DataT, Dims, Mode, target::device> : public detail::accessor_base
 				throw std::runtime_error(
 				    "Calling accessor constructor with device target is only allowed in parallel_for tasks."
 				    "If you want to access this buffer from within a host task, please specialize the call using one of the *_host_task tags");
 			}
 
-			auto& live_cgh = dynamic_cast<detail::live_pass_device_handler&>(cgh);
 			// It's difficult to figure out which stored range mapper corresponds to this constructor call, which is why we just call the raw mapper manually.
-			const auto mapped_sr = live_cgh.apply_range_mapper<Dims>(rmfn, buff.get_range());
-			const auto access_info =
-			    detail::runtime::get_instance().get_buffer_manager().access_device_buffer<DataT, Dims>(detail::get_buffer_id(buff), Mode, mapped_sr);
-
+			auto access_info =
+			    detail::runtime::get_instance().get_buffer_manager().access_device_buffer<DataT, 0>(detail::get_buffer_id(buff), Mode, subrange<0>());
 			m_device_ptr = static_cast<DataT*>(access_info.ptr);
-			m_index_offset = detail::id_cast<Dims>(access_info.backing_buffer_offset);
-			m_buffer_range = detail::range_cast<Dims>(access_info.backing_buffer_range);
 		}
 	}
 
-	size_t get_linear_offset(const id<Dims>& index) const { return detail::get_linear_index(m_buffer_range, index - m_index_offset); }
-};
+	template <access_mode TagModeNoInit>
+	accessor(const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<Mode, TagModeNoInit, target::device> /* tag */) : accessor(buff, cgh) {}
+
+	template <access_mode TagMode, access_mode M = Mode, typename = std::enable_if_t<detail::access::mode_traits::is_producer(M)>>
+	accessor(
+	    const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<TagMode, Mode, target::device> /* tag */, const property::no_init /* no_init */)
+	    : accessor(buff, cgh) {}
+
+	template <access_mode TagMode, access_mode TagModeNoInit>
+	accessor(const buffer<DataT, 0>& buff, handler& cgh, const detail::access_tag<TagMode, TagModeNoInit, target::device> /* tag */,
+	    const property_list& /* prop_list */) {
+		static_assert(detail::constexpr_false<std::integral_constant<access_mode, TagMode>>,
+		    "Currently it is not accepted to pass a property list to an accessor constructor. Please use the property celerity::no_init "
+		    "as a last argument in the constructor");
+	}
+
+	template <access_mode M = Mode>
+	std::enable_if_t<detail::access::mode_traits::is_producer(M), DataT&> operator*() const {
+		return *m_device_ptr;
+	}
+
+	template <access_mode M = Mode>
+	std::enable_if_t<detail::access::mode_traits::is_pure_consumer(M), const DataT&> operator*() const {
+		return *m_device_ptr;
+	}
 
-// Celerity Accessor Deduction Guides
-template <typename T, int D, typename Functor, typename TagT>
-accessor(const buffer<T, D>& buff, handler& cgh, Functor rmfn, TagT tag)
-    -> accessor<T, D, detail::deduce_access_mode<TagT>(), detail::deduce_access_target<std::remove_const_t<TagT>>()>;
```

Comment:
Theres still some stray forward declarations for these functions left on lines 38-45.

Reply:
Removed.